pkg/cache/v3/linear.go

```diff
@@ -42,13 +44,25 @@ type LinearCache struct {
 	watches map[string]watches
 	// Set of watches for all resources in the collection
 	watchAll watches
+	// Set of delta watches. A delta watch always contain the list of subscribed resources
+	// together with its current version
+	// version and versionPrefix fields are ignored for delta watches, because we always generate the resource version.
+	deltaWatches map[int64]DeltaResponseWatch
+	// Continously incremented counter used to index delta watches.
+	deltaWatchCount int64
```

Comment:
Do we want a watchCount for linear instead of a set of watches like sotw? I'm guessing it would be too difficult to keep that style of caching mechanism

Reply:
I took a closer look at the corresponding sotw implementation. So we use [map[string]map[chan Response]struct{}](https://github.com/s-matyukevich/go-control-plane/blob/incremental-linear/pkg/cache/v3/linear.go#L44) to store watches in sotw vs in the delta variant we use [map[int64]DeltaResponseWatch](https://github.com/s-matyukevich/go-control-plane/blob/incremental-linear/pkg/cache/v3/linear.go#L50) As far as I understand, we need to group watchers by resource name in the sotw implementation because [here](https://github.com/s-matyukevich/go-control-plane/blob/incremental-linear/pkg/cache/v3/linear.go#L143-L148) we use resource names to track what resources the watcher is subscribed to. We don't need to do this in the delta varian because we track what resources we subscribed to using information from the stream state [here](https://github.com/s-matyukevich/go-control-plane/blob/incremental-linear/pkg/cache/v3/delta.go#L54) That's why we don't need to additionally group by resource name in the delta implementation.

Another difference is that sotw implementation uses  `chan Response]` as the key in the map and delta uses int64, and this is the reason why we have `deltaWatchCount int64` field, so we can generate unique watch id for every new watch. We can use the same approach as sotw and use `chan DeltaResponse` as the map key or even `DeltaResponseWatch` object. I am not sure it worth it though, because the current implementation doesn't add a lot of complexity and additionally we can log every new watch id [here](https://github.com/s-matyukevich/go-control-plane/blob/incremental-linear/pkg/cache/v3/linear.go#L341), which could be useful.

Reply:
Okay cool, I was just curious on if it was even worth trying to follow the sotw linear implementation delta but it sounds like its not. 

I didn't think about how we use the stream state in delta vs sotw so we don't have to track the resource name in the watches set.